src/urllib3/__init__.py

```diff
@@ -2,6 +2,36 @@
 Python HTTP library with thread-safe connection pooling, file post support, user friendly, and more
 """
 
+import sys
+
+CURRENT_PYTHON = sys.version_info[:2]
+REQUIRED_PYTHON = (3, 6)
+
+# This check and everything above must remain compatible with Python 2.7.
+if CURRENT_PYTHON < REQUIRED_PYTHON:
+    sys.stderr.write(
+        """
+==========================
+Unsupported Python version
+==========================
+This version of urllib3 requires Python {}.{}, but you're trying to
+install it on Python {}.{}.
+This may be because you are using a version of pip that doesn't
+understand the python_requires classifier. Make sure you
+have pip >= 9.0 and setuptools >= 24.2, then try again:
+    $ python -m pip install --upgrade pip setuptools
+    $ python -m pip install urllib3
+This will install the latest version of urllib3 which works on your
+version of Python. If you can't upgrade your pip (or Python), request
+an older version of urllib3:
+    $ python -m pip install "urllib3<2"
+""".format(
+            *(REQUIRED_PYTHON + CURRENT_PYTHON)
+        )
+    )
+    sys.exit(1)
```

Comment:
I think `sys.exit` is probably the wrong approach. We can raise an exception to force things to stop. Why do we presume we want to always exit? That's making lots of dangerous assumptions that can really annoy users

Reply:
I also agree `sys.exit()` is too strong, was thinking `ImportError`?

Reply:
I'd rather not `ImportError` because that can be confused with something being missing. Instead I'd suggest something like `UnsupportedRuntimeVersion` to make it really obvious.